### PR TITLE
Adds HEAP ammo(awesome)

### PR DESCRIPTION
**Pull request by CapCamIII**
## About The Pull Request

Geeves said I could make this PR.
https://discord.com/channels/150315577943130112/964684928161808384/1047399005665103892

Adds HEAP ammo, the ammo used in-lore by pretty much everyone to all weapons that use 10x24, 10x20, and the Type71.
Adds HEAP ammo to Marine Raiders(MARSOC), Whiteout, UPP Officers, and UPP Commandos
Adds magazines, ammo type, ammo boxes, and magazine boxes for all HEAP magazines, except HPR/M40SD/NSG23, who only have loose round boxes.
Sprites for above.

HEAP Ammo Stats are as follows

Rifle:
55 Damage
40 AP
Unchanged other than suicide head-overlay(bigger because thats funny)

Heavy Rifle(Type71):
50 Damage
40 AP(not as much as normal AP, but slight balance)
Unchanged otherwise

SMG:
45 damage
30 AP
Same bullet speed as AP, cause its not HV anymore.
Bigger suicide overlay

## Why It's Good For The Game

REASONING: I am very much tired of admins trying to do the "HEAP ammo" but with the comedically OP explosive ammo(essentially an impact HEDP that travels at bullet speeds out of a 40 round rifle magazine), so I decided to code and sprite this, it is OP, yes, but it is SIGNIFICANTLY less overpowered than the commonly used in events explosive ammo, and its also cool to have in-game.

Onto each point from about the pr section:

Adds HEAP ammo/magazines to every 10x24/10x20/Type71 weapon. 
Allow HEAP to be spawned if desired for events, for any marine weapon except shotguns, as well as a few non-marine weapons(type71/NSG).

Gives it to the Deathsquad-tier ERTs(UPP Commandos,, Whiteout, MARSOC)
UPP Commandos were weak, this gives them a nice buff, putting them to their "deathsquad" ert-level; the other two enhances their position as "don't send these guys unless you want the round to end" and makes them more of a threat to hostiles(and themselves)

Gives it to UPP Officers
Thought it would be a nice addition, self-defense ammo for them, as they're admin only and special, but not too attached to this if it needs to be removed.

Magazines
Required

Ammo boxes/magazine boxes
Easier spawning

Sprites
Required
 

## Changelog

:cl:
add: Adds HEAP ammo, does more damage than standard bullets of its type, but has AP bullet AP, a direct upgrade, but admin-only.
add: Adds magazines and ammo boxes that use HEAP ammo, in their respective calibers and bullet-types, NSG23, MK1, MK2, M39, Type71, HPR, M40SD, and L42A
balance: Marine Raiders, Whiteout, UPP officers, and UPP commandos are now equipped with HEAP ammo
imageadd: Added icons for HEAP Type71, M40SD, M39, MK1, NSG23, L42A, and HPR
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/pouch.dm [hunk 1]
```diff
@@ -539,6 +543,15 @@
...
 
+/obj/item/storage/pouch/magazine/large/rifle_heap/fill_preset_inventory()
+	for(var/i = 1 to storage_slots)
+		new /obj/item/ammo_magazine/rifle/heap(src)
+
+/obj/item/storage/pouch/magazine/large/smg_heap/fill_preset_inventory()
+	for(var/i = 1 to storage_slots)
+		new /obj/item/ammo_magazine/smg/m39/heap(src)
+
+
```
Comment (harryob) [inline code/game/objects/items/storage/pouch.dm:554]:
```suggestion
```
lol

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -215,6 +215,9 @@
 	scatter_unwielded = SCATTER_AMOUNT_TIER_4
 	damage_mult = BASE_BULLET_DAMAGE_MULT + BULLET_DAMAGE_MULT_TIER_5
 
+/obj/item/weapon/gun/rifle/m41a/elite/whiteout
+	current_mag = /obj/item/ammo_magazine/rifle/heap//special version for whiteout that has the HEAP mag, nothing else changed
```
Comment (harryob) [inline code/modules/projectiles/guns/rifles.dm:219]:
```suggestion
	current_mag = /obj/item/ammo_magazine/rifle/heap //special version for whiteout that has the HEAP mag, nothing else changed
```

File: code/modules/projectiles/magazines/rifles.dm [hunk 3]
```diff
@@ -34,6 +34,12 @@
 	icon_state = "m41a_explosive"
 	default_ammo = /datum/ammo/bullet/rifle/explosive
 
+/obj/item/ammo_magazine/rifle/heap
+	name = "\improper M41A HEAP magazine (10x24mm)"
+	desc = "A 10mm armor piercing high explosive magazine"
```
Comment (harryob) [inline code/modules/projectiles/magazines/rifles.dm:39]:
```suggestion
	desc = "A 10mm armor piercing high explosive magazine."
```
consistency

File: code/modules/projectiles/magazines/smgs.dm [hunk 4]
```diff
@@ -36,6 +36,12 @@
 	icon_state = "m39_le"
 	default_ammo = /datum/ammo/bullet/smg/rubber
 
+/obj/item/ammo_magazine/smg/m39/heap
+	name = "\improper M39 HEAP magazine (10x20mm)"
+	desc = "A 10x20mm caseless armor-piercing high-explosive submachinegun magazine. The bullet tips are made out of a special explosive, designed to penetrate armor then detonate for maximum soft-tissue damage."
```
Comment (harryob) [inline code/modules/projectiles/magazines/smgs.dm:41]:
```suggestion
	desc = "A 10x20mm caseless armor-piercing high-explosive submachinegun magazine. The bullet tips are made out of a special explosive material, designed to penetrate armor then detonate for maximum soft-tissue damage."
```
this makes more sense to me but if it's actually dumb just don't change it

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/magazines/smgs.dm (hunk 4 above)
Comment (CapCamIII) [reply code/modules/projectiles/magazines/smgs.dm:41]: the tip would realistically have more than just the explosive material, arming mechanism etc, so yeah probably dumb.

**Review by ThePiachu — APPROVED** [review]
Code looks good

File: code/modules/projectiles/magazines/smgs.dm (hunk 4 above)
Comment (harryob) [reply code/modules/projectiles/magazines/smgs.dm:41]: i can't believe this happened to me

**Review by harryob — APPROVED** [review]

**Review by nauticall — CHANGES_REQUESTED** [review]
honestly the dark blue coloring for the HEAP blends in far too much with the black of the magazines themselves and is really hard to distinguish from a distance; I'd either recommend it be made lighter (which can be done) _or_ make a new color entirely like yellow or something.

**Review by nauticall — APPROVED** [review]
yeah the yellow looks better now, approved!

**Pull request merged**